### PR TITLE
CXX-1476 Fix broken key lookup and add unit tests

### PR DESCRIPTION
**Pull request by acmorrow**
Subsumes PR #607

### CODE REVIEW

**Comment by acmorrow** [comment]
@hanumantmk PTAL. Might want to see CXX-992 as well for background.

**Comment by acmorrow** [comment]
@xdg @mbroadst @hanumantmk - This is a blocker for 3.2.0-rc1. Could one of you please take a look at it ASAP?

**Comment by xdg** [comment]
@saghm PTAL

File: src/mongocxx/test/collection.cpp [hunk 1]
```diff
@@ -2336,7 +2336,7 @@ TEST_CASE("create_index tests", "[collection]") {
 
         bool unique = options.unique().value();
         auto validate = [unique](bsoncxx::document::view index) {
-            auto expire_after = index["expireAfter"];
+            auto expire_after = index["expireAfterSeconds"];
```
Comment (saghm) [inline src/mongocxx/test/collection.cpp]: Did the previous version only work due to the bug being fixed above, or is this change unrelated?

Comment (acmorrow) [reply src/mongocxx/test/collection.cpp]: Only worked due to the bug.

**Comment by acmorrow** [comment]
@saghm ping?

**Pull request merged**